src/Cake.Core/Scripting/IScriptConfiguration.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+
+namespace Cake.Core.Scripting
+{
+    /// <summary>
+    /// Provides the ability to configure the script via IoC.
+    /// </summary>
+    public interface IScriptConfiguration
+    {
+        /// <summary>
+        /// Gets additional namespaces to be imported in the script.
+        /// </summary>
+        IReadOnlyList<string> Namespaces { get; }
+
+        /// <summary>
+        /// Gets additional assembly names to be referenced by the script.
+        /// </summary>
+        IReadOnlyList<string> AssemblyNames { get; }
```

Comment:
Since order isn't important here, can we change this to `IReadOnlyCollection<string>`?